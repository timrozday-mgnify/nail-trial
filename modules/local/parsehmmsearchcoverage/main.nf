process PARSEHMMSEARCHCOVERAGE {
    tag "${meta.id}"
    label 'process_single'

    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:0.1.1--pyhdfd78af_0'
        : 'biocontainers/mgnify-pipelines-toolkit:0.1.1--pyhdfd78af_0'}"

    input:
    tuple val(meta), path(domtbl_file), path(db)

    output:
    tuple val(meta), path(out_fp), emit: tsv
    tuple val(meta), path(stats_out_fp), emit: stats
    path "versions.yml", emit: versions

    script:
    def args = task.ext.args ?: ''
    out_fp = "${meta.id}_pfam_coverage.tsv"
    stats_out_fp = "${meta.id}_stats.json"

    def script = file("${moduleDir}/bin/hmmer_domtbl_parse_coverage.py")

    """
    python ${script} ${args} -i ${domtbl_file} -o ${out_fp} -s ${stats_out_fp} -d ${db}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        parsehmmsearchcoverage: \$(python --version |& sed '1!d ; s/python //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    out_fp = "${meta.id}_pfam_coverage.tsv"

    """
    touch ${out_fp}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        parsehmmsearchcoverage: \$(samtools --version |& sed '1!d ; s/samtools //')
    END_VERSIONS
    """
}
