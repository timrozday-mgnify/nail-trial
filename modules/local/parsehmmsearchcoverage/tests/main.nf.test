nextflow_process {
    name "Test Process PARSEHMMSEARCHCOVERAGE"
    script "../main.nf"
    process "PARSEHMMSEARCHCOVERAGE"

    tag "modules"
    tag "modules_"
    tag "parsehmmsearchcoverage"

    test("mini sample should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    file("${projectDir}/tests/fixtures/parsehmmsearchcoverage/ERR10889056_mini.domtbl.gz", checkIfExists: true),
		]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }

    test("edge-cases synthetic sample should match snapshot") {

        when {
            process {
                """
                input[0] = [
		    ['id': 'test'],
		    file("${projectDir}/tests/fixtures/parsehmmsearchcoverage/synthetic_edgecase_test.domtbl.gz", checkIfExists: true),
		]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}
