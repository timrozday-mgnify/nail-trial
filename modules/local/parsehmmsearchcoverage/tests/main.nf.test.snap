{
    "mini sample should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_pfam_coverage.tsv:md5,17f0ded0bfcb006a11005f7b0eb4dd91"
                    ]
                ],
                "1": [
                    "versions.yml:md5,65a450616ed2e91f2323322ba04e3b9b"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_pfam_coverage.tsv:md5,17f0ded0bfcb006a11005f7b0eb4dd91"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,65a450616ed2e91f2323322ba04e3b9b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T09:09:01.858120151"
    },
    "edge-cases synthetic sample should match snapshot": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_pfam_coverage.tsv:md5,e401f1728dde25c79d0beaeb87bc80ff"
                    ]
                ],
                "1": [
                    "versions.yml:md5,65a450616ed2e91f2323322ba04e3b9b"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_pfam_coverage.tsv:md5,e401f1728dde25c79d0beaeb87bc80ff"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,65a450616ed2e91f2323322ba04e3b9b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-05-20T09:12:40.333051066"
    }
}